### PR TITLE
fix: Sends 404 status code for invalid write path

### LINKED ISSUE #835: Wrong HTTP status code returned on write path
The write API is returning the wrong status code when a database can't be found:

```
⇒  curl -i "http://127.0.0.1:8080/api/v2/write?org=company&bucket=sensors" --data-binary 'cpu,server=a v=2.2'

HTTP/1.1 500 Internal Server Error
x-powered-by: Routerify v2.0.0-beta-2
content-length: 129
date: Fri, 19 Feb 2021 12:55:32 GMT

{"error":"Internal error writing points into org company, bucket sensors:  database not found: company_sensors","error_code":100}% 
```

It should be returning a `4xx` error (either `400` or `404` would make sense to me), not a `500`.

### PR DESCRIPTION
**Pull request by mitch292**
Closes #835

Sends a 400 for a database that doesn't exist rather than a 500. New to the project, let me know if any changes to the PR are needed!

Describe your proposed changes here.

- [X] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [X] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by e-dard — CHANGES_REQUESTED** [review]
Hey @mitch292 thanks so much for the contribution! I can see you have signed the CLA.

I think it would be good to add a test case for this if possible? Let us know if you need help with that.

Finally, your PR title needs to be in the semantic commit format. Take a look at closed PRs for examples of that.

**Comment by mitch292** [comment]
Hi just wanted to bump this PR. I added a test and fixed the name of the pr and branch. Let me know if I need to make any other changes!

File: src/influxdb_ioxd/http.rs [hunk 1]
```diff
@@ -205,7 +205,7 @@ impl ApplicationError {
         match self {
             Self::BucketByName { .. } => self.internal_error(),
             Self::BucketMappingError { .. } => self.internal_error(),
-            Self::WritingPoints { .. } => self.internal_error(),
+            Self::WritingPoints { .. } => self.bad_request(),
```
Comment (tustvold) [inline src/influxdb_ioxd/http.rs:208]: `WritingPoints` is returned for any error writing lines, with this change genuine internal errors will be returned as bad requests.

It should be possible to match on the error within `influxdb_ioxd::http::write` and convert a `server::Error::DatabaseNotFound` into a `influxdb_ioxd::http::ApplicationError::DatabaseNotFound`

Comment (mitch292) [reply src/influxdb_ioxd/http.rs]: Ah of course. Took a stab at this and updated the tests!

**Review by e-dard — APPROVED** [review]
This LGTM now! 👍

**Review by carols10cents — APPROVED** [review]
LGTM too! Thanks!

**Pull request merged**